src/main/java/org/example/src/game/Game.java

```diff
@@ -1,68 +1,93 @@
 package org.example.src.game;
 import org.example.src.units.*;
+
+
+import java.io.IOException;
 import java.util.Scanner;
 
 public  class Game {
-    protected static Hero hero ;
+    protected static Hero hero;
     protected static Unit enemy;
 
+    protected static SaveSystem saveSystem;
+
+    static {
+        try {
+            saveSystem = new SaveSystem();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
     protected static Scanner input = new Scanner(System.in);
-    protected static void setHero(String name){
+
+    protected static void setHero(String name) {
         hero = new Hero(name);
     }
-    protected static void setEnemy(Unit obj){
+
+    protected static void setEnemy(Unit obj) {
         enemy = obj;
     }
-    protected static void getAction(){
+
+    protected static void getAction() {
+        boolean save;
         upper:
-        while(true) {
+        while (true) {
+            save = false;
             int choice = input.nextInt();
             try {
                 switch (choice) {
-                    case 1 :{
+                    case 1: {
                         hero.giveHit(enemy);
                         break upper;
                     }
-                    case 2 :{
+                    case 2: {
                         hero.getHeal();
                         break upper;
                     }
-                    case 3 :{
+                    case 3: {
                         hero.giveUp();
                         break upper;
                     }
-                    default : {
+                    case 4: {
+                        saveSystem.save(enemy,hero);
+                        save = true;
```

Comment:
This variable is not necessary - it actually makes the code less readable. If you just want to save and continue the game, you might as well use the `continue ` keyword after saving. It will move you to the beginning of the loop then.